src/CoreWCF.NetTcp/tests/ServiceBehaviorUseSynchronizationContextAsyncTests.cs

```diff
@@ -0,0 +1,169 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.ServiceModel.Channels;
+using System.Threading;
+using System.Threading.Tasks;
+using Contract;
+using CoreWCF.Configuration;
+using Helpers;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Services;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace CoreWCF.NetTcp.Tests
+{
+    /// <summary>
+    /// Verifies the effect of <see cref="ServiceBehaviorAttribute.UseSynchronizationContext"/> on asynchronous operations.
+    /// </summary>
+    public class ServiceBehaviorUseSynchronizationContextAsyncTests
+    {
+        public const string RelativeAddress = "/nettcp.svc/UseSynchronizationContextAsync";
+        private readonly ITestOutputHelper _output;
+
+        public ServiceBehaviorUseSynchronizationContextAsyncTests(ITestOutputHelper output)
+        {
+            _output = output;
+
+            TaskService.s_synchronizationContextWhenCreated = null;
+            TaskService.s_synchronizationContextWhenCalled = null;
+            TaskService.s_managedThreadIdWhenCalled = null;
+
+            TaskServiceWithUseSynchronizationContextFalse.s_synchronizationContextWhenCreated = null;
+            TaskServiceWithUseSynchronizationContextFalse.s_synchronizationContextWhenCalled = null;
+        }
+
+        /// <summary>
+        /// This test exists only to demonstrate what the UIFact attribute does.
+        /// </summary>
+        [UIFact]
+        public async Task UIFactTestAsync()
+        {
+            int originalThread = Environment.CurrentManagedThreadId;
+            await Task.Yield();
+            Assert.Equal(originalThread, Environment.CurrentManagedThreadId);
+        }
+
+        [UIFact]
+        public async Task UseSynchronizationContextTrueWithSynchronizationContextTestAsync()
+        {
+            SynchronizationContext orgSynchronizationContext = SynchronizationContext.Current;
+            Assert.NotNull(orgSynchronizationContext); // Set by the UIFact attribute
+            int orgManagedThreadId = Environment.CurrentManagedThreadId;
+
+            await RunAsync<TaskService>();
+
+            Assert.Same(orgSynchronizationContext, TaskService.s_synchronizationContextWhenCreated);
+            Assert.Same(orgSynchronizationContext, TaskService.s_synchronizationContextWhenCalled);
+            Assert.Equal(orgManagedThreadId, TaskService.s_managedThreadIdWhenCalled);
+        }
+
+        [Fact]
+        public async Task UseSynchronizationContextTrueWithoutSynchronizationContextTestAsync()
+        {
+            await RunWithoutSynchronizationContextAsync(async () =>
+            {
+                Assert.Null(SynchronizationContext.Current);
+                int orgManagedThreadId = Environment.CurrentManagedThreadId;
+
```

Comment:
You aren't using orgManagerThreadId in the test validation. Is there missing validation or was this line left in place by accident?

Reply:
I had a validation there before and forgot to remove the variable. I added a comment now on why it wasn't ok, in case it helps someone in the future.